File: GoblinGardenGuards/goblins.py
def dist2(xx, x, yy, y, r2):
    return ((xx - x) ** 2 + (yy - y) ** 2) <= r2

def sprinkle(x, y, r):
    startX = max(0, x - r -1)
    endX = min(10000, x+ r + 1)
    startY = max(0, y - r-1)
    endY = min(10000, y + r + 1)
    r2 = r**2
    for xx in range(startX, endX):
        for yy in range(startY, endY):
            garden[xx][yy] = garden[xx][yy] or dist2(xx, x, yy, y, r2)


G = input()
goblins = []
for g in range(G):
    x, y = map(int, raw_input().split())
    goblins.append((x, y))
M =input()

garden = [[False] * 10001 for _ in range(10001)]

for m in range(M):
    x, y, r = map(int, raw_input().split())
    sprinkle(x, y, r)

c = 0
for x in range(10001):
    for y in range(10001):
        if garden[x][y]:
            c += 1
print c

count = 0
for x, y in goblins:
    if not garden[x][y]:
        count += 1
print count

File: GoblinGardenGuards/Goblins.java
import java.util.*;
import java.io.*;

public class Goblins{

    BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
    private int hasNext = 0;
    private String[] buff;

    public static void main(String[] args) throws Exception{
        new Goblins().run();
    }
    private void fail() {
        throw new IllegalArgumentException();
    }

    private void print(String s){
        System.out.println(s);
    }


    private void print(int s){
        System.out.println(s);
    }

    private int readInt() throws Exception{

        if(hasNext == 0){
          buff = (reader.readLine()).split(" ");
        }
        int n = Integer.parseInt(buff[hasNext]);
        hasNext = (hasNext + 1) % buff.length;
        return n;
    }

    private String readStr() throws Exception{

        if(hasNext == 0){
          buff = (reader.readLine()).split(" ");
        }
        String n = buff[hasNext];
        hasNext = (hasNext + 1) % buff.length;
        return n;
    }

    private boolean dist2(int xx, int x, int yy, int y, int r2){
        int dx = (xx - x);
        int dy = (yy - y);
        return (dx * dx + dy * dy) <= r2;
      }

      private void sprinkle(int x, int y, int r){

            int startX = Math.max(0, x - r-1);
            int endX = Math.min(10001, x+ r + 1);
            int startY = Math.max(0, y - r-1);
            int endY = Math.min(10001, y + r + 1);
            int r2 = r *r;
            for(int xx = startX; xx < endX; xx++){
              for(int yy = startY; yy < endY; yy++){
                  garden[xx][yy] = garden[xx][yy] || dist2(xx, x, yy, y, r2);
              }
            }
      }


    boolean[][] garden;

    private void run() throws Exception{
            int G = readInt();
            int[][] goblins = new int[G][2];
            for(int i = 0; i < G;i++){
              int x = readInt();
              int y = readInt();
              goblins[i] = new int[]{x, y};
            }
            int M =readInt();
            int lim = 10001;
            garden = new boolean[lim][lim];
            for(int i = 0; i < M;i++){
              int x = readInt();
              int y = readInt();
              int r = readInt();
              sprinkle(x, y, r);
            }
            int count = 0;
            for(int i = 0; i < G;i++){
              int x  = goblins[i][0];
              int y  = goblins[i][1];

              if(!garden[x][y]){
                count += 1;
              }
            }
            print(count);
    }
}
/*


*/
